packages/cli/medusa-cli/src/reporter/index.ts

```diff
@@ -174,27 +175,57 @@ export class Reporter {
    * Level 0
    */
   error(messageOrError: string | Error, error?: Error) {
-    let message = messageOrError as string
+    let message: string
+    let errorAsObject: Error | undefined
 
-    if (typeof messageOrError === "object") {
+    if (messageOrError && typeof messageOrError === "object") {
+      errorAsObject = messageOrError
       message = messageOrError.message
-      error = messageOrError
+    } else if (error) {
+      message = messageOrError
+      errorAsObject = error
+    } else {
+      message = messageOrError
     }
 
     const toLog = {
       level: "error",
       message,
     }
 
-    if (error) {
-      toLog["stack"] = stackTrace.parse(error)
+    if (errorAsObject) {
+      toLog["message"] = errorAsObject.message
+      toLog["stack"] = stackTrace.parse(errorAsObject)
+      /**
+       * Winston only logs the error properties when they are
+       * string values. Hence we will have to self convert
+       * the error cause to a string
+       */
+      if ("cause" in errorAsObject) {
+        toLog["cause"] = inspect(errorAsObject.cause)
+      }
     }
 
+    /**
+     * If "errorAsObject" has a message property, then we will
+     * print the standalone message as one log item and then
+     * the actual error object as another log item.
+     *
+     * Otherwise, we always loose the message property from the
+     * actual error object
+     */
+    if (errorAsObject?.message && errorAsObject?.message !== message) {
+      this.loggerInstance_.log({ level: "error", message })
+    }
     this.loggerInstance_.log(toLog)
 
-    // Give stack traces and details in dev
-    if (error && IS_DEV) {
-      console.log(error)
+    /**
+     * In dev we print the error using `console.error`. I am not sure if
+     * we should be doing this, because now we are giving more info
+     * in Dev and less info in production
+     */
+    if (errorAsObject && IS_DEV) {
+      console.error(errorAsObject)
```

Comment:
discussion: I am not sure about that either, also why are we suddenly using the console API there? I think it should be similar no matter the env right?

Reply:
Yeah. But I noticed that Winston doesn't display the error stack when not using the `JSON` formatter. 

This is how it looks when I remove the `console.error` statement

![CleanShot 2024-11-06 at 13 59 31@2x](https://github.com/user-attachments/assets/ed2acadc-6131-48b6-8a14-f30f6a26de5f)